### ISSUE TITLE
(BREAKING CHANGES) Refactoring MobileAds API to v3.0.0

### ISSUE THREAD

**Issue by gumaciel** [Poing-Studios/godot-admob-plugin#82]
### Is your feature request related to a problem? If so, please describe.

This change intends to change many of the plugins in all aspects, in order to make the plugin more dynamic and easy to use, it is necessary almost completely to redo the plugin to meet the use of mediation.

⚠️WARNING⚠️:
- No long support for Godot 3 due to the high complexity of refactoring (but there's the [godot3](https://github.com/Poing-Studios/godot-admob-editor/tree/godot3) branch such as the supported versions in the [release tab](https://github.com/Poing-Studios/godot-admob-editor/releases))
- Next tag release will be v3.0.0, when Godot 5 arrives probably will be v4.0.0, and create a branch to store the old version.
- Will be release with Android and iOS version

### What feature or improvement would you like to see?

✅GOALS✅
- [x] Create an `Interface` of MobileAds class with GDScript with the most important methods used on AdMob. 

Example:

```gdscript
extends Control

func _ready() -> void:
	MobileAds.initialize(OnInitializationCompleteListener.new(_on_initialization_complete))

func _on_initialization_complete(initialization_status : InitializationStatus) -> void:
	for key in initialization_status.adapter_status_map:
		var adapterStatus : AdapterStatus = initialization_status.adapter_status_map[key]
		prints(key, adapterStatus.latency, adapterStatus.initialization_status, adapterStatus.description)
```

Android documentation: https://developers.google.com/admob/android/mediate#java
```java
import com.google.android.gms.ads.MobileAds;
import com.google.android.gms.ads.initialization.AdapterStatus;
import com.google.android.gms.ads.initialization.InitializationStatus;
import com.google.android.gms.ads.initialization.OnInitializationCompleteListener;

public class MainActivity extends AppCompatActivity {

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        MobileAds.initialize(this, new OnInitializationCompleteListener() {
            @Override
            public void onInitializationComplete(InitializationStatus initializationStatus) {
                Map<String, AdapterStatus> statusMap = initializationStatus.getAdapterStatusMap();
                for (String adapterClass : statusMap.keySet()) {
                    AdapterStatus status = statusMap.get(adapterClass);
                    Log.d("MyApp", String.format(
                            "Adapter name: %s, Description: %s, Latency: %d",
                            adapterClass, status.getDescription(), status.getLatency()));
                }

                // Start loading ads here...
            }
        });
    }
}
```
- [x] https://github.com/Poing-Studios/godot-admob-editor/issues/23
- [x] Add the possibility of having as many mediations as possible, but their total development may take longer, in the beginning only one or two mediations will be developed
- [x] Dynamic creation of advertisements, you can load several advertisements at once (should attend https://github.com/Poing-Studios/godot-admob-android/issues/133)
- [x] Change the name of the repository from `godot-admob-editor` to `godot-admob-plugin`
- [ ] https://github.com/Poing-Studios/godot-admob-ios/issues/45
- [ ] https://github.com/Poing-Studios/godot-admob-android/issues/167

Other goals could be increased during the development

### Additional context

Should be in the next minor release (e.g.: v3.1.0):
-  Creational of Unit Tests to have a little more assurance the plugin will work well (probably use [gdUnit4](https://github.com/MikeSchulze/gdUnit4))

### Release Planning, Version Support, and Versioning:
https://github.com/Poing-Studios/godot-admob-plugin/discussions/87

### Calendar Version Support:
https://github.com/Poing-Studios/godot-admob-plugin/discussions/88

**Comment by Momo90k** [comment]
Is mediation finally supported?
If yes, can I use the plugin with Godot version 3.5.+?
Thank you for your work.

**Comment by gumaciel** [comment]
Hi @Momo90k , it will be supported when this issue is closed and merged on `master` branch

This refactoring aims to Godot v4.1.

This should work on 3.5+, but you will need to write usage of this plugin in Godot in your way, when this issue closes, it will be all ready for 4.1.

**Comment by Momo90k** [comment]
Ok. Thank you.
